cmd/registry/push/push.go

```diff
@@ -127,6 +147,30 @@ func (o *pushOptions) RunPush(ctx context.Context, args []string) error {
 		return fmt.Errorf("an error occurred while creating the pusher for registry %s: %w", registry, err)
 	}
 
+	// Make sure to remove temporary working dir.
+	defer func() {
+		if err := os.RemoveAll(toBeDeleted); err != nil {
+			o.Printer.Warning.Printfln("Unable to remove temporary dir %q: %s", toBeDeleted, err.Error())
+		}
+	}()
+
+	for i, p := range paths {
+		if err = utils.IsTarGz(filepath.Clean(p)); err != nil && !errors.Is(err, utils.ErrNotTarGz) {
+			return err
+		} else if err == nil {
+			continue
+		} else {
+			path, err := utils.CreateTarGzArchive(p)
+			if err != nil {
+				return err
+			}
+			paths[i] = path
+			if toBeDeleted == "" {
```

Comment:
What happens if multiple `paths` need to be archived? Are we going to cleanup only the first one?

Reply:
Uh i see, we store the folder, not the filepath. It's ok then, sorry!